src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD106UseInvokeAsyncForAsyncEventsAnalyzer.cs

```diff
@@ -65,27 +59,24 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
 
-            context.RegisterCodeBlockStartAction<SyntaxKind>(ctxt =>
+            context.RegisterOperationBlockStartAction(context =>
             {
-                // This is a very specical case to check if this method is TplExtensions.InvokeAsync().
+                // This is a very special case to check if this method is TplExtensions.InvokeAsync().
                 // If it is, then do not run the analyzer inside that method.
-                if (!(ctxt.OwningSymbol.Name == Types.TplExtensions.InvokeAsync &&
-                      ctxt.OwningSymbol.ContainingType.Name == Types.TplExtensions.TypeName &&
-                      ctxt.OwningSymbol.ContainingType.BelongsToNamespace(Types.TplExtensions.Namespace)))
+                if (!(context.OwningSymbol.Name == Types.TplExtensions.InvokeAsync &&
+                      context.OwningSymbol.ContainingType.Name == Types.TplExtensions.TypeName &&
+                      context.OwningSymbol.ContainingType.BelongsToNamespace(Types.TplExtensions.Namespace)))
                 {
-                    ctxt.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(this.AnalyzeInvocation), SyntaxKind.InvocationExpression);
+                    context.RegisterOperationAction(Utils.DebuggableWrapper(this.AnalyzeInvocation), OperationKind.Invocation);
                 }
             });
         }
 
-        /// <summary>
-        /// Analyze each invocation syntax node.
-        /// </summary>
-        private void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
+        private void AnalyzeInvocation(OperationAnalysisContext context)
         {
-            var invocation = (InvocationExpressionSyntax)context.Node;
-            var symbol = context.SemanticModel.GetSymbolInfo(invocation.Expression, context.CancellationToken).Symbol;
-            if (symbol != null)
+            var invocation = (IInvocationOperation)context.Operation;
+            var symbol = invocation.TargetMethod;
+            if (symbol is object)
```

Comment:
symbol is guaranteed to be non-null

Reply:
also the symbol kind check below is redundant - it will always be a method.

Reply:
It'll be great when this whole API is null ref annotated to make these guarantees more obvious. :)

Reply:
Removed the null check